plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPageSinkProvider.java

```diff
@@ -38,12 +39,12 @@ public JdbcPageSinkProvider(JdbcClient jdbcClient)
     @Override
     public ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorOutputTableHandle tableHandle)
     {
-        return new JdbcPageSink(session, (JdbcOutputTableHandle) tableHandle, jdbcClient);
+        return new JdbcPageSink(session, (JdbcOutputTableHandle) tableHandle, jdbcClient, getInsertBatchSize(session));
     }
 
     @Override
     public ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorInsertTableHandle tableHandle)
     {
-        return new JdbcPageSink(session, (JdbcOutputTableHandle) tableHandle, jdbcClient);
+        return new JdbcPageSink(session, (JdbcOutputTableHandle) tableHandle, jdbcClient, getInsertBatchSize(session));
```

Comment:
The PageSink has the session available already - so can we initialise the batchSize in the contructor instead of injecting it?

@kokosing WDYT?

Reply:
That make sense to me

Reply:
done